s3_resource/include/irods/private/s3_resource/multipart_shared_data.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef S3_RESOURCE__MULTIPART_SHARED_DATA_HPP
+#define S3_RESOURCE__MULTIPART_SHARED_DATA_HPP
+
+#include <boost/interprocess/managed_shared_memory.hpp>
+#include <boost/interprocess/containers/map.hpp>
+#include <boost/interprocess/containers/vector.hpp>
+#include <boost/interprocess/containers/list.hpp>
+#include <boost/interprocess/allocators/allocator.hpp>
+#include <boost/interprocess/managed_shared_memory.hpp>
+#include <boost/interprocess/containers/string.hpp>
+#include <boost/interprocess/sync/named_mutex.hpp>
+#include <boost/container/scoped_allocator.hpp>
+#include <boost/interprocess/sync/scoped_lock.hpp>
+
+#include <iostream>
+
+namespace irods_s3 
+{
+    namespace interprocess_types
+    {
+        namespace bi          = boost::interprocess;
+        using segment_manager = bi::managed_shared_memory::segment_manager;
+        using void_allocator  = boost::container::scoped_allocator_adaptor<bi::allocator<void, segment_manager> >;
+    }
+
+    // data that needs to be shared among different processes
+    struct multipart_shared_data
+    {
+        explicit multipart_shared_data(const interprocess_types::void_allocator &allocator)
+            : threads_remaining_to_close{0}
+            , number_of_threads{0}
+            , ref_count{0}
```

Comment:
Should `ref_count` be initialized to 1 instead of 0?

Reply:
This object is created via the named_shared_memory_object.  That constructor increases the ref_count so 0 is appropriate.

In reset_fields, the object already exists so it made sense for this to be set to 1.  However, reset_fields() is no longer used so I have removed that method from both the resource and transport.

Reply:
Got it.